tiles/src/main/java/com/protomaps/basemap/names/OsmNames.java

```diff
@@ -21,14 +22,46 @@ public static FeatureCollector.Feature setOsmNames(FeatureCollector.Feature feat
       var script = Script.getScript(value);
 
       if (key.equals("name")) {
-        feature.setAttrWithMinzoom("name", value, minZoom);
+        List<String> segments = ScriptSegmenter.segmentByScript(value);
+        if (segments.size() >= 1) {
+          int index = 0;
+          feature.setAttrWithMinzoom("name", segments.get(index), minZoom);
 
-        if (!script.equals("Latin") && !script.equals("Generic")) {
-          feature.setAttrWithMinzoom("pmap:script", script, minZoom);
+          script = Script.getScript(segments.get(index));
+
+          if (!script.equals("Latin") && !script.equals("Generic")) {
+            feature.setAttrWithMinzoom("pmap:script", script, minZoom);
+          }
+
+          String encodedValue = TextEngine.encodeRegisteredScripts(segments.get(index));
+          feature.setAttrWithMinzoom("pmap:pgf:name", encodedValue, minZoom);
```

Comment:
The prefix is needed because when writing style expressions, you don't know if `name` contains Devanagari or doesn't. For v4 I would like to propose that we switch it the other way round, store pgf encoded text in `name` and have an option to store bare unicode in an additional property such as `original:name` for example.

Reply:
I'm weary of making every name tag gibberish, that's not great ergonomics for dev or end users when most name tags being Unicode works fine today, with many benefits thru the entire tile cut and (generic/universal) map display stack. For example, I can view tiles in QGIS now, but messing with the Unicode encoding would break that.

Can you elaborate more about why you find the proposed solution compelling?

Reply:
Only text in scripts that are not supported by MapLibre are affected, such as for example Devanagari. Latin, Greek, etc will be identical to what we have now.

From my point of view the vector tiles should only contain the data that is needed to render the Protomaps basemaps styles in MapLibre. We can facilitate tile generation time switches to include more that that can be useful for other purposes.

Reply:
> The prefix is needed because when writing style expressions, you don't know if name contains Devanagari or doesn't. 

To restate, currently in main the label for Switzerland has `pmap:pgf:name = Schweiz/Suisse/Svizzera/Svizra`, is there a way to omit this and only have `pmap:pgf:name` for unsupported scripts?

Reply:
We can if we use a coalesce expression whenever we access a name property in the style. The required change looks like this: https://github.com/protomaps/basemaps/pull/275/commits/fba63a6b80e331bf928680ab87a6950790c25950